overlord/snapstate/autorefresh.go

```diff
@@ -644,11 +644,22 @@ func (m *autoRefresh) launchAutoRefresh() error {
 		return err
 	}
 
-	if _, err := createPreDownloadChange(m.state, updateTss); err != nil {
+	createdPreDl, err := createPreDownloadChange(m.state, updateTss)
+	if err != nil {
 		return err
 	}
 
 	if len(updateTss.Refresh) == 0 {
+		if createdPreDl {
+			// NOTE: If all refresh candidates are blocked from auto-refresh by checks
+			// in softCheckNothingRunningForRefresh then no auto-refresh change will be
+			// created (i.e. len(updateTss.Refresh) == 0) and only a pre-download change
+			// is created for those snaps. This still means that some snaps are being
+			// inhibited from auto-refresh so we are recording a notice about it here
+			// because it cannot be captured in processInhibitedAutoRefresh which only
+			// looks for auto-refresh changes.
+			return maybeAddRefreshInhibitNotice(m.state)
```

Comment:
which means this maybe needs to uncoditional now? and always be called if createPrDl is true or false

Reply:
Now that `maybeAddRefreshInhibitNotice` checks last recorded inhibited snaps set, the condition is a no-op. I removed it, Thank you!